### ISSUE TITLE
[FEA] Implement iloc

### ISSUE THREAD

**Issue by mrocklin** [rapidsai/cudf#606]
It might be useful to implement iloc, at least on rows.  

```python
import cudf
df = cudf.DataFrame({'x': range(10)})
df.iloc[:5]
```

```python-traceback
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-a92f47027042> in <module>
----> 1 df.iloc[:5]

~/cudf/python/cudf/dataframe/dataframe.py in __getattr__(self, key)
    180             return self[key]
    181
--> 182         raise AttributeError("'DataFrame' object has no attribute %r" % key)
    183
    184     def __getitem__(self, arg):

AttributeError: 'DataFrame' object has no attribute 'iloc'
```

**Comment by AK-ayush** [comment]
That is definitely useful api. I have started working on it...

**Comment by kkraus14** [comment]
Fixed by #608
